### DISCUSSION TITLE
Issue with cut_with_mesh() deleting vertices unexpectedly

### DISCUSSION THREAD

**Discussion by Linus-Foley** [marcomusy/vedo#1193] [General]
I have a U shaped object, and I wish to cut off a section of one of its ends. I'm trying to do this by placing a rectangle at the place I want to slice at, and then using cut_with_mesh() using the rectangle. As you can see below, the opposite side of the U shape gets some of its vertices deleted.

![image](https://github.com/user-attachments/assets/039912da-29b6-4176-8e36-cea395fa9c79)

Is there a way to prevent this, or another way to go about cutting off one end of a U shape?

**Comment by marcomusy** [comment]
Sorry for the late reply,
When you use cut_with_plane() the Plane is defined in 3d space by its normal so it's an infinite plane.
Consider using cut_with_box() or cut_with_mesh() instead.

**Reply by Linus-Foley** [reply]
Hi Marco, this bug is occurring with cut_with_mesh(), not cut_with_plane().

If it's useful, here's the shapes to reproduce the problem. I'm just doing `half_torus.cut_with_mesh(rectangle, invert=True)`

[shapes.zip](https://github.com/user-attachments/files/17466730/shapes.zip)

**Comment by marcomusy** [comment]
Oh ok, that is because there is no "inside" of a mesh when the mesh is a plane.
Either try with cut_with_box() or e.g.:
```py
from vedo import *

torus = Mesh("half_torus.stl")
m = Mesh("rectangle.stl").c("blue5")
box = m.clone().rotate_x(90).extrude(-0.2).rotate_x(-90)
box.compute_normals()
box.c("green").alpha(0.2).lw(1).wireframe()

torus.cut_with_mesh(box, invert=1)
show(torus, m, box, axes=1)
```
![image](https://github.com/user-attachments/assets/884c2797-791c-4a4b-be24-b9047e934eb1)

**Reply by Linus-Foley** [reply]
Nice, thanks!
